core/templates/dev/head/galleries/Gallery.js

```diff
@@ -201,6 +204,115 @@ oppia.controller('Gallery', [
       }
 
       $rootScope.loadingMessage = '';
+
+      // The following initializes the gallery carousel(s).
+      // Pause is necessary to ensure all elements have loaded.
+      $timeout(initCarousels, 500);
+
+      if (typeof $scope.galleryGroups != 'undefined') {
+        // The following initializes the tracker to have all
+        // elements flush left.
+        cleanLoad = false;
+        for (i = 0; i < $scope.galleryGroups.length; i++) {
+          $scope.rowIndices[i] = 0;
+        }
+      }
+    });
+
+    // Below is the width of each tile.
+    var TILE_DISPLAY_WIDTH = 211.953;
+    // If the value below is changed, the following CSS values in oppia.css
+    // must be changed:
+    // - .oppia-gallery-tiles-carousel: max-width
+    // - .oppia-gallery-tiles-container: max-width
+    var MAX_TILE_DISPLAY = 4;
+    $scope.tileDisplayCount = 0;
+    $scope.galleryTilesResizedWidth = 0;
+
+    var initCarousels = function() {
+      var windowWidth = $(window).width() * 0.85;
+      $scope.tileDisplayCount = Math.floor(windowWidth / TILE_DISPLAY_WIDTH);
+      $scope.galleryTilesResizedWidth = $scope.tileDisplayCount *
+        TILE_DISPLAY_WIDTH;
+      if ($scope.galleryTilesResizedWidth >
+          TILE_DISPLAY_WIDTH * MAX_TILE_DISPLAY) {
+        $scope.galleryTilesResizedWidth = TILE_DISPLAY_WIDTH * MAX_TILE_DISPLAY;
+        $scope.tileDisplayCount = MAX_TILE_DISPLAY;
+      }
+      $('.oppia-gallery-tiles-carousel').css({
+        width: $scope.galleryTilesResizedWidth + 'px'
+      });
+
+      // The following determines whether to enable left scroll after resize.
+      for (i = 0; i < $('.oppia-gallery-tiles-carousel').length; i++) {
```

Comment:
Instead of referring to the jQuery selector, is it possible to just use the length of $scope.galleryGroups?